src/vellum/workflows/state/context.py

```diff
@@ -156,23 +157,47 @@ def resolve_workflow_deployment(
         Returns:
             BaseWorkflow instance if found, None otherwise
         """
-        if not self.generated_files or not self.namespace:
+        if not self._generated_files or not self._namespace:
             return None
 
         expected_prefix = generate_workflow_deployment_prefix(deployment_name, release_tag)
 
-        workflow_file_key = f"{expected_prefix}/workflow.py"
-        if workflow_file_key not in self.generated_files:
-            return None
-
         try:
             from vellum.workflows.workflows.base import BaseWorkflow
 
             WorkflowClass = BaseWorkflow.load_from_module(f"{self.namespace}.{expected_prefix}")
+            WorkflowClass.is_dynamic = True
             workflow_instance = WorkflowClass(context=WorkflowContext.create_from(self), parent_state=state)
             return workflow_instance
         except Exception:
-            return None
+            pass
+
+        try:
+            major_version = __version__.split(".")[0]
+            version_range = f">={major_version}.0.0,<={__version__}"
+
+            response = self.vellum_client.workflows.pull(
+                deployment_name=deployment_name,
+                release_tag=release_tag,
+                version=version_range,
+            )
```

Comment:
cc: @m-abboud are there any concerns at a networking/alert level with most subworkflow deployment nodes getting to this point and returning a 422 before this gets fully rolled out?

Reply:
Yes this will likely trigger alerts, the only 4xx we don't alert on is 429